cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1152,45 +1158,93 @@ void writer::impl::write(table_view const& table)
   // compression/decompression performance).
   using cudf::io::parquet::gpu::max_page_fragment_size;
 
-  size_type const num_fragments = (num_rows + max_page_fragment_size - 1) / max_page_fragment_size;
+  bool row_group_sizes_specified = !row_group_sizes.empty();
+  size_type num_fragments        = (num_rows + max_page_fragment_size - 1) / max_page_fragment_size;
```

Comment:
Is this the same math as `div_rounding_up_safe`? That utility function might be applicable in a few places here.